roles/helm-ansible-base/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+##############################################################################
+# Provision helm-ansible-base
+#
+# This role uses helm and kubectl to provision or deprovision the resources
+# defined by a helm chart.
+##############################################################################
+
+- block:
+  - name: create values file
+    tempfile:
+      state: file
+    register: values_file
+
+  - name: create overrides file
+    tempfile:
+      state: file
+    register: overrides_file
+
+  - name: write content to values file
+    copy:
+      content: "{{ values }}"
+      dest: "{{ values_file.path }}"
+
+  - name: render overrides template
+    template:
+      src: "{{ hb_overrides_template }}"
+      dest: "{{ overrides_file.path }}"
+
+  - name: run helm template
+    shell: "helm template --debug --name {{ hb_release_name }} -f {{ values_file.path }} -f {{ overrides_file.path }} /opt/chart.tgz | sed -n '/---/,$p' > {{ hb_manifest_path }}"
+    register: helmout
+    failed_when: helmout.rc != 0
+
+  - debug:
+      msg: "{{ helmout.stdout }}"
+
+  - debug:
+      msg: "{{ helmout.stderr }}"
+
+  - name: run kubectl
+    shell: "kubectl {{ hb_k8s_action }} -n {{ namespace }} -f {{ hb_manifest_path }}"
+    register: kubectlout
+    when: hb_apply == true
+    failed_when: (kubectlout.rc != 0 and hb_k8s_action == "create")
+
+  when: hb_k8s_action == "create"
```

Comment:
I agree with fabian's [comment on my PR ](https://github.com/automationbroker/automation-broker-apb/pull/1#discussion_r180826245). Give the block a `name` even though it isn't rendered and add the `when` at the top for readability.